packages/exhibitor/extra/dcos_zk_backup.py

```diff
@@ -0,0 +1,358 @@
+#!/usr/bin/env python3
+"""
+Command line tool for automated DC/OS ZooKeeper instance backup and restore.
+
+This script is made available as `dcos-zk` executable in DC/OS.
+
+"""
+import shlex
+import shutil
+import subprocess
+import sys
+import tarfile
+from argparse import ArgumentParser, ArgumentTypeError
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import Optional
+
+
+EXHIBITOR_DIR = Path('/var/lib/dcos/exhibitor')
+
+
+def run_command(cmd: str, verbose: bool) -> None:
+    """
+    Run a command in a subprocess.
+
+    Args:
+        verbose: Show the output.
+
+    Raises:
+        subprocess.CalledProcessError: The given cmd exits with a non-0 exit code.
+    """
+    stdout = None if verbose else subprocess.PIPE
+    stderr = None if verbose else subprocess.STDOUT
+    subprocess.run(
+        args=shlex.split(cmd),
+        stdout=stdout,
+        stderr=stderr,
+        check=True,
+    )
+
+
+def _is_zookeeper_running(verbose: bool) -> bool:
+    """
+    Exhibitor and therefore ZooKeeper is in active state (running).
+    """
+    try:
+        run_command('systemctl status dcos-exhibitor', verbose)
```

Comment:
I think this is OK, but there is a dedicated way to tell if service is running

https://unix.stackexchange.com/questions/396630/the-proper-way-to-test-if-a-service-is-running-in-a-script
https://www.freedesktop.org/software/systemd/man/systemctl.html#is-active%20PATTERN%E2%80%A6